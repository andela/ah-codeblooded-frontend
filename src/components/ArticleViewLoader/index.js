import React from 'react';
import ContentLoader from 'react-content-loader';

export default props => (
  <ContentLoader
    rtl
    height={500}
    width={400}
    speed={3}
    primaryColor="#cccccc"
    secondaryColor="#ecebeb"
    {...props}
  >
    <rect x="196.03" y="33.61" rx="0" ry="0" width="190" height="120" />
    <rect x="7.03" y="36.61" rx="0" ry="0" width="163" height="11" />
    <rect x="9.03" y="54.61" rx="0" ry="0" width="160" height="11" />
    <rect x="9.03" y="76.61" rx="0" ry="0" width="72" height="6" />
    <circle cx="22.82" cy="110.4" r="13.79" />
    <rect x="43.03" y="100.61" rx="0" ry="0" width="31" height="4" />
    <rect x="44.03" y="107.61" rx="0" ry="0" width="21" height="4" />
    <rect x="15.03" y="137.61" rx="0" ry="0" width="20.85" height="19.46" />
    <rect x="46.03" y="139.61" rx="0" ry="0" width="63" height="17" />
    <rect x="26.03" y="194.61" rx="0" ry="0" width="357" height="8" />
    <rect x="26.03" y="212.61" rx="0" ry="0" width="356" height="7" />
    <rect x="27.03" y="231.61" rx="0" ry="0" width="356" height="8" />
    <rect x="26.03" y="257.61" rx="0" ry="0" width="356" height="109" />
    <rect x="28.03" y="379.61" rx="0" ry="0" width="348" height="4" />
    <rect x="27.03" y="395.61" rx="0" ry="0" width="351" height="7" />
    <rect x="28.03" y="415.61" rx="0" ry="0" width="349" height="6" />
    <rect x="27.03" y="435.61" rx="0" ry="0" width="348" height="6" />
    <rect x="27.03" y="455.61" rx="0" ry="0" width="349" height="6" />
    <rect x="27.03" y="476.61" rx="0" ry="0" width="347" height="6" />
  </ContentLoader>
);
