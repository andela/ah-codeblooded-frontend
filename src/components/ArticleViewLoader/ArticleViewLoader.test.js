import React from 'react';
import { shallow } from 'enzyme';
import ArticleViewLoader from './index';

describe('<ArticleLoader/> component', () => {
  it('renders <ArticleLoader/> component without crashing', () => {
    const snapShot = shallow(<ArticleViewLoader />);
    expect(snapShot).toMatchSnapshot();
  });
});
