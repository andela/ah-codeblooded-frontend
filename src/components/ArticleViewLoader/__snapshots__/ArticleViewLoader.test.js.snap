// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleLoader/> component renders <ArticleLoader/> component without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <rect
          height="180"
          rx="5"
          ry="5"
          width="496"
          x="2"
          y="2"
        />,
        <rect
          height="33.35"
          rx="0"
          ry="0"
          width="285.36"
          x="8.03"
          y="209.61"
        />,
        <rect
          height="0"
          rx="0"
          ry="0"
          width="0"
          x="198.03"
          y="194.61"
        />,
        <rect
          height="3"
          rx="0"
          ry="0"
          width="482"
          x="7.03"
          y="259.61"
        />,
        <rect
          height="0"
          rx="0"
          ry="0"
          width="0"
          x="176.03"
          y="208.61"
        />,
        <rect
          height="10"
          rx="0"
          ry="0"
          width="0"
          x="179.03"
          y="209.61"
        />,
        <rect
          height="3"
          rx="0"
          ry="0"
          width="482"
          x="5.03"
          y="279.61"
        />,
        <rect
          height="3"
          rx="0"
          ry="0"
          width="482"
          x="6.03"
          y="299.61"
        />,
        <rect
          height="3"
          rx="0"
          ry="0"
          width="482"
          x="8.03"
          y="318.61"
        />,
        <rect
          height="3"
          rx="0"
          ry="0"
          width="482"
          x="9.03"
          y="338.61"
        />,
        <rect
          height="3"
          rx="0"
          ry="0"
          width="482"
          x="6.03"
          y="399.61"
        />,
        <rect
          height="3"
          rx="0"
          ry="0"
          width="482"
          x="8.03"
          y="419.61"
        />,
        <rect
          height="3"
          rx="0"
          ry="0"
          width="482"
          x="8.03"
          y="438.61"
        />,
        <rect
          height="3"
          rx="0"
          ry="0"
          width="482"
          x="9.03"
          y="458.61"
        />,
        <rect
          height="3"
          rx="0"
          ry="0"
          width="289.2"
          x="8.03"
          y="359.61"
        />,
        <rect
          height="3"
          rx="0"
          ry="0"
          width="289.2"
          x="9.03"
          y="479.61"
        />,
      ],
      "height": 500,
      "primaryColor": "#cccccc",
      "rtl": true,
      "secondaryColor": "#ecebeb",
      "speed": 3,
      "width": 500,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "height": "180",
          "rx": "5",
          "ry": "5",
          "width": "496",
          "x": "2",
          "y": "2",
        },
        "ref": null,
        "rendered": null,
        "type": "rect",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "height": "33.35",
          "rx": "0",
          "ry": "0",
          "width": "285.36",
          "x": "8.03",
          "y": "209.61",
        },
        "ref": null,
        "rendered": null,
        "type": "rect",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "height": "0",
          "rx": "0",
          "ry": "0",
          "width": "0",
          "x": "198.03",
          "y": "194.61",
        },
        "ref": null,
        "rendered": null,
        "type": "rect",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "height": "3",
          "rx": "0",
          "ry": "0",
          "width": "482",
          "x": "7.03",
          "y": "259.61",
        },
        "ref": null,
        "rendered": null,
        "type": "rect",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "height": "0",
          "rx": "0",
          "ry": "0",
          "width": "0",
          "x": "176.03",
          "y": "208.61",
        },
        "ref": null,
        "rendered": null,
        "type": "rect",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "height": "10",
          "rx": "0",
          "ry": "0",
          "width": "0",
          "x": "179.03",
          "y": "209.61",
        },
        "ref": null,
        "rendered": null,
        "type": "rect",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "height": "3",
          "rx": "0",
          "ry": "0",
          "width": "482",
          "x": "5.03",
          "y": "279.61",
        },
        "ref": null,
        "rendered": null,
        "type": "rect",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "height": "3",
          "rx": "0",
          "ry": "0",
          "width": "482",
          "x": "6.03",
          "y": "299.61",
        },
        "ref": null,
        "rendered": null,
        "type": "rect",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "height": "3",
          "rx": "0",
          "ry": "0",
          "width": "482",
          "x": "8.03",
          "y": "318.61",
        },
        "ref": null,
        "rendered": null,
        "type": "rect",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "height": "3",
          "rx": "0",
          "ry": "0",
          "width": "482",
          "x": "9.03",
          "y": "338.61",
        },
        "ref": null,
        "rendered": null,
        "type": "rect",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "height": "3",
          "rx": "0",
          "ry": "0",
          "width": "482",
          "x": "6.03",
          "y": "399.61",
        },
        "ref": null,
        "rendered": null,
        "type": "rect",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "height": "3",
          "rx": "0",
          "ry": "0",
          "width": "482",
          "x": "8.03",
          "y": "419.61",
        },
        "ref": null,
        "rendered": null,
        "type": "rect",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "height": "3",
          "rx": "0",
          "ry": "0",
          "width": "482",
          "x": "8.03",
          "y": "438.61",
        },
        "ref": null,
        "rendered": null,
        "type": "rect",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "height": "3",
          "rx": "0",
          "ry": "0",
          "width": "482",
          "x": "9.03",
          "y": "458.61",
        },
        "ref": null,
        "rendered": null,
        "type": "rect",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "height": "3",
          "rx": "0",
          "ry": "0",
          "width": "289.2",
          "x": "8.03",
          "y": "359.61",
        },
        "ref": null,
        "rendered": null,
        "type": "rect",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "height": "3",
          "rx": "0",
          "ry": "0",
          "width": "289.2",
          "x": "9.03",
          "y": "479.61",
        },
        "ref": null,
        "rendered": null,
        "type": "rect",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <rect
            height="180"
            rx="5"
            ry="5"
            width="496"
            x="2"
            y="2"
          />,
          <rect
            height="33.35"
            rx="0"
            ry="0"
            width="285.36"
            x="8.03"
            y="209.61"
          />,
          <rect
            height="0"
            rx="0"
            ry="0"
            width="0"
            x="198.03"
            y="194.61"
          />,
          <rect
            height="3"
            rx="0"
            ry="0"
            width="482"
            x="7.03"
            y="259.61"
          />,
          <rect
            height="0"
            rx="0"
            ry="0"
            width="0"
            x="176.03"
            y="208.61"
          />,
          <rect
            height="10"
            rx="0"
            ry="0"
            width="0"
            x="179.03"
            y="209.61"
          />,
          <rect
            height="3"
            rx="0"
            ry="0"
            width="482"
            x="5.03"
            y="279.61"
          />,
          <rect
            height="3"
            rx="0"
            ry="0"
            width="482"
            x="6.03"
            y="299.61"
          />,
          <rect
            height="3"
            rx="0"
            ry="0"
            width="482"
            x="8.03"
            y="318.61"
          />,
          <rect
            height="3"
            rx="0"
            ry="0"
            width="482"
            x="9.03"
            y="338.61"
          />,
          <rect
            height="3"
            rx="0"
            ry="0"
            width="482"
            x="6.03"
            y="399.61"
          />,
          <rect
            height="3"
            rx="0"
            ry="0"
            width="482"
            x="8.03"
            y="419.61"
          />,
          <rect
            height="3"
            rx="0"
            ry="0"
            width="482"
            x="8.03"
            y="438.61"
          />,
          <rect
            height="3"
            rx="0"
            ry="0"
            width="482"
            x="9.03"
            y="458.61"
          />,
          <rect
            height="3"
            rx="0"
            ry="0"
            width="289.2"
            x="8.03"
            y="359.61"
          />,
          <rect
            height="3"
            rx="0"
            ry="0"
            width="289.2"
            x="9.03"
            y="479.61"
          />,
        ],
        "height": 500,
        "primaryColor": "#cccccc",
        "rtl": true,
        "secondaryColor": "#ecebeb",
        "speed": 3,
        "width": 500,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "height": "180",
            "rx": "5",
            "ry": "5",
            "width": "496",
            "x": "2",
            "y": "2",
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "height": "33.35",
            "rx": "0",
            "ry": "0",
            "width": "285.36",
            "x": "8.03",
            "y": "209.61",
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "height": "0",
            "rx": "0",
            "ry": "0",
            "width": "0",
            "x": "198.03",
            "y": "194.61",
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "height": "3",
            "rx": "0",
            "ry": "0",
            "width": "482",
            "x": "7.03",
            "y": "259.61",
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "height": "0",
            "rx": "0",
            "ry": "0",
            "width": "0",
            "x": "176.03",
            "y": "208.61",
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "height": "10",
            "rx": "0",
            "ry": "0",
            "width": "0",
            "x": "179.03",
            "y": "209.61",
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "height": "3",
            "rx": "0",
            "ry": "0",
            "width": "482",
            "x": "5.03",
            "y": "279.61",
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "height": "3",
            "rx": "0",
            "ry": "0",
            "width": "482",
            "x": "6.03",
            "y": "299.61",
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "height": "3",
            "rx": "0",
            "ry": "0",
            "width": "482",
            "x": "8.03",
            "y": "318.61",
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "height": "3",
            "rx": "0",
            "ry": "0",
            "width": "482",
            "x": "9.03",
            "y": "338.61",
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "height": "3",
            "rx": "0",
            "ry": "0",
            "width": "482",
            "x": "6.03",
            "y": "399.61",
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "height": "3",
            "rx": "0",
            "ry": "0",
            "width": "482",
            "x": "8.03",
            "y": "419.61",
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "height": "3",
            "rx": "0",
            "ry": "0",
            "width": "482",
            "x": "8.03",
            "y": "438.61",
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "height": "3",
            "rx": "0",
            "ry": "0",
            "width": "482",
            "x": "9.03",
            "y": "458.61",
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "height": "3",
            "rx": "0",
            "ry": "0",
            "width": "289.2",
            "x": "8.03",
            "y": "359.61",
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "height": "3",
            "rx": "0",
            "ry": "0",
            "width": "289.2",
            "x": "9.03",
            "y": "479.61",
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
